modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java

```diff
@@ -196,10 +198,42 @@ public static BigDecimal toBigDecimal(Object o, int precision, int scale) {
 
         if (o instanceof Boolean) {
             throw new UnsupportedOperationException();
+        } else if (o instanceof Number) {
+            return toBigDecimal((Number) o, precision, scale);
+        } else {
+            return toBigDecimal(o.toString(), precision, scale);
```

Comment:
do we really want to try to transform any types here ?

Reply:
Reverted these changes (but it was just formatting) since original code does the same.